pages/recording/[id].tsx

```diff
@@ -103,22 +108,31 @@ function useRecordingSlug(recordingId: string) {
 
 function RecordingPage({
   getAccessibleRecording,
+  setExpectedError,
   head,
 }: PropsFromRedux & { head?: React.ReactNode }) {
   const store = useStore();
   const recordingId = useGetRecordingId();
+  const rawRecordingId = useGetRawRecordingIdWithSlug();
```

Comment:
This two are quite similar and confusing when used at the same time; we could probably unify them into something that returns a single object with e.g. `{ id: string, idAndSlug: string, slug: string }` etc.

Also, considering that a bunch of functions that live inside `hooks/recordings.ts` have `get` in their name and actually do a remote fetch, the naming of these functions is then more wrong, so I'd suggest renaming to something along the lines get `useQueryRecordingMeta` or `useRouterRecordingMeta` or similar, to better express the type of operation (sync or async) and where the data is coming from (router vs store vs remote gql).

Reply:
In general, we could probably do something more smart about organizing recording information and hooks that fetch required data.

We have a single recording, and parts of its data are either there, or require fetching. Wrapping everything inside a single hook along the lines of `useCurrentRecording` would handle a bunch of responsibility of calling the right functions, eliminate a ton of smaller unnecessary hooks and simplify the interface, centralising it in one place. But indeed out of scope for this PR particularly.